modules/instrument_builder/jsx/react.questions.js

```diff
@@ -481,7 +481,9 @@ class DateOptions extends Component {
             <select
               id="dateFormat"
               className="form-control"
-              onChange={this.onChange}>
+              onChange={this.onChange}
+              defaultValue={dateFormat}
```

Comment:
defaultValue sets the type of date to show correctly when editing the date field.